### PR TITLE
feat: Extend DTR APIs for Access management - Add local file implementation of the service

### PR DESCRIPTION
**Pull request by istvan-nagy-epam**
## Description

- Adds new module to implement service contract for access control rules
- Adds new tests for the service and repository implementations

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

File: access-control-service-sql-impl/src/main/java/org/eclipse/tractusx/semantics/accesscontrol/sql/service/SqlBackedAccessControlRuleService.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+import org.eclipse.tractusx.semantics.accesscontrol.sql.repository.AccessControlRuleRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class SqlBackedAccessControlRuleService implements AccessControlRuleService {
+
+   private final AccessControlRuleRepository repository;
+
+   public SqlBackedAccessControlRuleService( @Autowired AccessControlRuleRepository repository ) {
```
Comment (tunacicek) [inline access-control-service-sql-impl/src/main/java/org/eclipse/tractusx/semantics/accesscontrol/sql/service/SqlBackedAccessControlRuleService.java:44]: You can remove the annotation @Autowired.

File: access-control-service-sql-impl/src/main/java/org/eclipse/tractusx/semantics/accesscontrol/sql/model/AccessRulePolicy.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            .collect( Collectors.toSet() );
+   }
+
+   private String getStringValueOfRule( final String ruleName ) {
+      return accessRules.stream().filter( rule -> ruleName.equals( rule.attribute() ) )
+            .map(  idValue -> getAccessRulePolicyValueStringFunction( idValue, ruleName ) )
+            .findAny().orElse( null );
+   }
+
+   private static String getAccessRulePolicyValueStringFunction( final AccessRulePolicyValue idValue, final String ruleName ) {
```
Comment (tunacicek) [inline access-control-service-sql-impl/src/main/java/org/eclipse/tractusx/semantics/accesscontrol/sql/model/AccessRulePolicy.java:91]: Is there a reason to make the methods static?

File: access-control-service-sql-impl/src/main/java/org/eclipse/tractusx/semantics/accesscontrol/sql/service/SqlBackedAccessControlRuleService.java (hunk 1 above)
Comment (istvan-nagy-epam) [reply access-control-service-sql-impl/src/main/java/org/eclipse/tractusx/semantics/accesscontrol/sql/service/SqlBackedAccessControlRuleService.java:44]: Removing, thank you!

File: access-control-service-sql-impl/src/main/java/org/eclipse/tractusx/semantics/accesscontrol/sql/model/AccessRulePolicy.java (hunk 2 above)
Comment (istvan-nagy-epam) [reply access-control-service-sql-impl/src/main/java/org/eclipse/tractusx/semantics/accesscontrol/sql/model/AccessRulePolicy.java:91]: No reason, it was just something I missed when the IDE extracted the method as static.

**Review by tunacicek — APPROVED** [review]
LGTM

**Pull request merged**